File: src/management_prd/models/bug.py
"""Bug 管理数据模型。

Bug 与需求（:class:`management_prd.models.requirement.RequirementItem`）分离存储在
独立的 ``bugs`` 表中。每条 bug 记录：

- 必属一个项目（``project_id``，级联删除）。
- 必填 ``module_names``（多模块，取自该项目 ``modules`` 表；后端 ``BugService`` 校验）。
- 必填 ``level``（P0-P4）。
- 有生命周期 ``status``（待修复 / 已修复）。
- 可选 ``linked_iteration_id`` 关联某条需求迭代（指向 ``requirements.id``），
  留空表示无关联。**不加外键**，关联失效（对应需求被删）由应用层 staleness 检测。
- ``date`` 为 bug 日期（用于时间聚合分组，与需求 ``date`` 口径一致）。

模块关联由 ``bug_modules`` 表表达（多对多），``BugItem.modules`` 为非持久化字段，
服务层在序列化前回填。
"""

from __future__ import annotations

import datetime
from enum import StrEnum

from pydantic import BaseModel


class BugLevel(StrEnum):
    """Bug 严重级别。值与前端 TypeScript 字面量共享同一份语义。"""

    P0 = "P0"  # 核心缺陷
    P1 = "P1"  # Critical
    P2 = "P2"  # High
    P3 = "P3"  # Medium（迁移与新建默认）
    P4 = "P4"  # Low


class BugStatus(StrEnum):
    """Bug 生命周期状态。"""

    OPEN = "open"  # 待修复（默认）
    FIXED = "fixed"  # 已修复


class BugItem(BaseModel):
    """一条 bug 记录。"""

    id: str
    project_id: str
    content: str
    level: BugLevel
    status: BugStatus = BugStatus.OPEN
    linked_iteration_id: str | None = None
    date: datetime.date
    created_at: datetime.datetime
    updated_at: datetime.datetime
    modules: list[str] = []  # 非持久化，API 层回填


class CreateBugInput(BaseModel):
    """新建 bug 入参。

    ``module_names`` 多模块（≥1），须为项目已有模块或待新建模块（服务层 ``ensure_modules``
    落表）。``linked_iteration_id`` 可空。
    """

    module_names: list[str]
    content: str
    level: BugLevel
    status: BugStatus = BugStatus.OPEN
    linked_iteration_id: str | None = None
    date: datetime.date


class UpdateBugInput(BaseModel):
    """更新 bug 入参（部分字段）。

    ``module_names`` 为 None 表示跳过；提供则整体替换关联。``linked_iteration_id`` 与
    ``clear_linked`` 配合实现三态（镜像
    :class:`management_prd.models.data.UpdateRequirementInput` 的 completion_deadline）：
    - linked_iteration_id=None, clear_linked=False -> 跳过
    - linked_iteration_id=<id> -> 设为该迭代
    - clear_linked=True -> 置 NULL（优先级高于 linked_iteration_id）
    """

    module_names: list[str] | None = None
    content: str | None = None
    level: BugLevel | None = None
    status: BugStatus | None = None
    linked_iteration_id: str | None = None
    clear_linked: bool = False
    date: datetime.date | None = None

File: frontend/src/types/bug.ts
/** Bug 管理相关类型定义（与后端 models/bug.py 共享语义）。 */

export type BugLevel = 'P0' | 'P1' | 'P2' | 'P3' | 'P4'
export type BugStatus = 'open' | 'fixed'

export const LEVEL_LABEL: Record<BugLevel, string> = {
  P0: 'P0 核心缺陷',
  P1: 'P1 Critical',
  P2: 'P2 High',
  P3: 'P3 Medium',
  P4: 'P4 Low',
}

export const LEVEL_TAG_TYPE: Record<BugLevel, string> = {
  P0: 'danger',
  P1: 'danger',
  P2: 'warning',
  P3: 'info',
  P4: 'success',
}

export const BUG_STATUS_LABEL: Record<BugStatus, string> = {
  open: '待修复',
  fixed: '已修复',
}

export const BUG_STATUS_TAG_TYPE: Record<BugStatus, string> = {
  open: 'danger',
  fixed: 'success',
}

/** 一条 bug 记录。 */
export interface BugItem {
  id: string
  project_id: string
  module: string
  content: string
  level: BugLevel
  status: BugStatus
  linked_iteration_id: string | null
  date: string // ISO yyyy-MM-dd
  created_at: string
  updated_at: string
}

/** 新建 bug 入参。 */
export interface CreateBugInput {
  module: string
  content: string
  level: BugLevel
  status: BugStatus
  linked_iteration_id: string | null
  date: string
}

/** 更新 bug 入参（部分字段）。 */
export interface UpdateBugInput {
  module?: string
  content?: string
  level?: BugLevel
  status?: BugStatus
  linked_iteration_id?: string
  /** true 则置 NULL（优先级高于 linked_iteration_id） */
  clear_linked?: boolean
  date?: string
}

/** resolve_bug_link 返回（用于详情页关联卡片 + 跨视图跳转四步）。 */
export interface BugLinkInfo {
  item_id: string
  project_id: string
  module: string
  feature: string
  content: string
  date: string
}
